org-code-javabuilder/lib/src/main/java/org/code/javabuilder/CodeRunner.java

```diff
@@ -10,6 +10,7 @@
    *
    * @param urlClassLoader class loader to load compiled code
    * @throws JavabuilderException if there is an error running code
+   * @return true if there was code to run, false if there was not
    */
-  void run(URLClassLoader urlClassLoader) throws JavabuilderException;
+  boolean run(URLClassLoader urlClassLoader) throws JavabuilderException;
```

Comment:
The return value here feels a little like code smell from a side effect. Given that the mainRunner ignores the return value, would it make more sense to have something like testRunner.hasTests() followed by testRunner.run()?

Reply:
I think that would overly complicate things. We don't know whether we have tests until we have done all the setup work to run them, so we would either do that work twice or enforce that the caller has to call the methods in a specific order, which I don't like. I think a run method with a boolean return value of did or did not run code is logical. For main we end up throwing exceptions in every case we fail.

Reply:
👍